src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxNodeExtensions.cs

```diff
@@ -26,10 +29,45 @@ internal static class SyntaxNodeExtensions
                 {
                     continue;
                 }
-                node = syntaxGenerator.AddAttributes(node, syntaxGenerator.Attribute(attribute));
 
+                if (IsReservedAttribute(attribute.AttributeClass))
+                {
+                    continue;
+                }
+
+                node = syntaxGenerator.AddAttributes(node, syntaxGenerator.Attribute(attribute));
             }
             return node;
         }
+
+        private static HashSet<string> _reservedTypes = new HashSet<string>(StringComparer.Ordinal)
+            {
+                "DynamicAttribute",
+                "IsReadOnlyAttribute",
+                "IsUnmanagedAttribute",
+                "IsByRefLikeAttribute",
+                "TupleElementNamesAttribute",
+                "NullableAttribute",
+                "NullableContextAttribute",
+                "NullablePublicOnlyAttribute",
+                "NativeIntegerAttribute",
+                "ExtensionAttribute",
+                "RequiredMemberAttribute",
+                "ScopedRefAttribute",
+                "RefSafetyRulesAttribute"
+            };
```

Comment:
I assume that we today don't have any "reserved" assembly attributes that should be filtered out? Asking as this code is only for "member" attributes.

Reply:
There are some that apply to the module, but not the assembly.  I'll add this to the assembly attribute case for consistency though.